convoy/batch.py

```diff
@@ -4471,41 +4470,75 @@ def _construct_task(
                             ),
                             '-npernode {}'.format(processes_per_node)
                         ])
-                    mpi_opts.extend(task.multi_instance.mpi.options)
+                if is_singularity:
+                    mpi_singularity_cmd = 'singularity {} {} {} {}'.format(
+                        task.singularity_cmd,
+                        ' '.join(task.run_options),
+                        task.singularity_image,
+                        task.command)
                     mpi_command = 'mpirun {} {}'.format(
                         ' '.join(mpi_opts),
-                        'singularity {} {} {} {}'.format(
-                            task.singularity_cmd,
-                            ' '.join(task.run_options),
-                            task.singularity_image,
-                            task.command
-                        )
+                        mpi_singularity_cmd
+                    )
+                else:
```

Comment:
add in-line comment after else regarding what conditional this refers to since the block is large